packages/embarkjs/src/storage.js

```diff
@@ -72,37 +72,27 @@ Storage.isAvailable = function () {
 };
 
 // TODO: most of this logic should move to the provider implementations themselves
-Storage.setProviders = async function (dappConnOptions) {
+Storage.setProviders = function (dappConnOptions) {
```

Comment:
Wow, this function is now very painful. Anyway we could convert it to only use promises with async/await or only use callbacks?

Reply:
Agreed that would make it more readable. But just to clarify, it was always really painful haha, and what may have looked easier to read was  actually hiding the reason for the error.

Reply:
Yes of course, not blaming you, just stating that this is ugly hahaha.
And hoping you had an idea on how to make it less ugly